Userland/DevTools/HackStudio/HackStudioWidget.cpp

```diff
@@ -1610,8 +1610,8 @@ HackStudioWidget::ContinueDecision HackStudioWidget::warn_unsaved_changes(String
 {
     if (!any_document_is_dirty())
         return ContinueDecision::Yes;
-
-    auto result = GUI::MessageBox::show(window(), prompt, "Unsaved changes"sv, GUI::MessageBox::Type::Warning, GUI::MessageBox::InputType::YesNoCancel);
+    auto dialog = verify_cast<HackStudioWidget>(window()->main_widget())->m_new_project_dialog;
```

Comment:
we are in HackStudioWidget here, you can just access this via `m_new_project_dialog` directly

Reply:
Of course.